Tribler/community/tunnel/hidden_community.py

```diff
@@ -312,26 +397,34 @@ def on_key_request(self, messages):
 
     def check_key_response(self, messages):
         for message in messages:
-            if not message.source.startswith(u"circuit_"):
```

Comment:
It now uses the same circuit as it received the key-request from, I really tried to understand why these circuit_ prefixes in the code did not work out as they should, but the on-key-response did not contain any code responsible for relaying to some circuit. A separate circuit for the keys makes it much easier to understand the code. It gives some overhead - true - but guarantees the message will always arrive. 

One thing that's not clear to me now, but you know better about this stuff then me: how can the intro-point relay a message to the downloader without a circuit between them (as the key-request is 'exited' over an exit-node circuit?)